### ISSUE TITLE
doesn't work with remote files

### ISSUE THREAD

**Issue by tweiu** [dinhquochan/nova-phplint#1]
no action if I worked with remote files via SFTP

**Comment by dinhquochan** [comment]
Yes, can you resolve this issue!? I don't see API for remote files. Feel free to open the PR.

**Comment by tweiu** [comment]
> Yes, can you resolve this issue!? I don't see API for remote files. Feel free to open the PR.

I found a possible fault: need to request rights to read filesystem. extension.json:
"entitlements": {
...
        "filesystem": "readonly",
...
    }

**Comment by tweiu** [comment]
instead of filepath passed to 'php -l' should use stdin and file content

`
const filePath = editor.document.path;
const phpPath = '/usr/local/bin/php'; // YOUR PHP PATH
const range = new Range( 0, editor.document.length );
const contents = editor.document.getTextInRange( range );
const { stdout, status } = await execute(phpPath, {args: ['-l']}, contents);
`

**Comment by tweiu** [comment]
and changes to execute, add third argument (string) and write to stdin after process.start():

let writer = process.stdin.getWriter();
writer.write( string );
writer.close();
